### ISSUE TITLE
FastSpeech 2 does not seem to be learning much from my dataset

### ISSUE THREAD

**Issue by PedroDKE** [TensorSpeech/TensorFlowTTS#500]
I'm trying to train a FastSpeech2 model on a dataset i scraped myself (around 7 hours of speech and 5000 audio files resampled to 22050hz). I do not expect great results, but at least something to work with. My preprocessing steps are similar to all the ones in the repository, except i had to put my min trimming db to 10.  Besides this i used the default preprocessing/normalizing steps and when i check my files they seem to look okay-ish (not great, but the recordings weren't made in a professional studio). some samples (MFCC, raw energy and raw f0) can be seen in a previous issue #496 . My durations aren't perfect at the moment but i think theyre fine for some first training/testing. 
After 200k iterations on my dataset i see that the model did not learn much. The durations seem to be okay, but the model struggles to learn f0 and energy. During the validation step i added my own custom plots that also plot the gt and predicted values of these. i also use the fastspeech2.v2.yaml configuration to have less parameters to train (but v1 seemed to have the same issue so i think it might be related to my data), and changed the dataset length to fit the character length of my own in tensorflow_tts/configs/fastspeech.py.

From my tensorboard results i can see that the f0 and energy loss in the evaluation are very unstable compared to the duration. Training losses seem to be fine i think, but when i try to generate speech from a sentence seen during training the model's performance is very very poor (unrecognisable speech) and similar to what i see in my validation.
![image](https://user-images.githubusercontent.com/43861296/108402938-74420080-721e-11eb-80b1-874a89fee98c.png)

As can be seen from these 2 images from my validation set@150k steps, the model did learn some of the MFCC features but they're represented very poorly. When i look at the f0 and energy plots, they are way off and my model does not seem to be learning these the right way. 

first example:
![b'9-00012-f000125'](https://user-images.githubusercontent.com/43861296/108403230-cdaa2f80-721e-11eb-8fda-ae030fa40f65.png)
![b'9-00012-f000125'_estimations](https://user-images.githubusercontent.com/43861296/108403238-cedb5c80-721e-11eb-897c-694fa7ac9e42.png)

second example:
![b'9-00018-f000015'](https://user-images.githubusercontent.com/43861296/108403827-6f318100-721f-11eb-9ae4-41abc0db47f6.png)
![b'9-00018-f000015'_estimations](https://user-images.githubusercontent.com/43861296/108403835-7062ae00-721f-11eb-8ab8-c48b8e813acc.png)

**Comment by ZDisket** [comment]
How did you extract your durations? MFA? Tacotron? Your data shouldn't be a trouble since I was able to make good sounding models out of 4 and 6 hours of 44.1KHz data.

**Comment by ZDisket** [comment]
Oops I misclicked sorry, open another issue

**Comment by PedroDKE** [comment]
I scaped them from librivox and alligned them to the book text by using aeneas. The allignments seem to be correct for 95%+ of my data that i'm using in this case.

**Comment by ZDisket** [comment]
@PedroDKE Are your durations from the aligner or from Tacotron?

**Comment by PedroDKE** [comment]
@ZDisket the aeneas alligner only works on sentence level. So I trained my own tacatron2 on the data to extract durations. I have talked about a bit in another issue (these are worst case scenarios, in most cases the durations cover the whole sentence, not perfect but i would assume it would be good enough to get some preliminary results) https://github.com/TensorSpeech/TensorFlowTTS/issues/496#issuecomment-779798995

**Comment by ZDisket** [comment]
@PedroDKE 
Can't help you that much since I use phoneme and MFA, but I had a similar problem in the past with some multispeaker datasets and I fixed it by changing the learning rate schedule to the one in the LibriTTS example: https://github.com/TensorSpeech/TensorFlowTTS/blob/master/examples/fastspeech2_libritts/conf/fastspeech2libritts.yaml

**Comment by PedroDKE** [comment]
@ZDisket my dataset is only a single speaker atm, but i plan to extend it to multiple speakers, I will try out the learning rate schedule, thank you! Is there anything you would recommend to try out if this doesnt work?

btw, i tried to use MFA too but i found out that on the character level, it does not return me a duration for each character ? for example when i have 40 characters (including punctuation) i only get like 36 duration values? this is the main reason why i did not continue to use MFA for my durations, but if this isnt a problem i can see if i can use this again.

**Comment by ZDisket** [comment]
@PedroDKE If you use MFA you want to use it for phoneme durations and train a model based on phonetic input, you can find extraction instructions: https://github.com/TensorSpeech/TensorFlowTTS/tree/master/examples/mfa_extraction, although I don't use those scripts specifically.

**Comment by PedroDKE** [comment]
@ZDisket so i adjusted the learning rate as you said, but i'm not sure if i can see any improvements.
Here is my tensorboard
![image](https://user-images.githubusercontent.com/43861296/108496658-ab5df380-72aa-11eb-8857-09562db44694.png)

and here is the first example again at approx 145k steps. I'm not sure if the model is better or not. But the MFCC's produced still aren't really usefull (compare to other tensorboards ive seen around in this github, my eval f0/energy losses also seem to be pretty high, they should be around 0.1 more i think ?)

![b'9-00012-f000125'](https://user-images.githubusercontent.com/43861296/108497059-350dc100-72ab-11eb-8958-eb0c176ef790.png)
![b'9-00012-f000125'_estimations](https://user-images.githubusercontent.com/43861296/108497067-37701b00-72ab-11eb-98e1-e32e024ddb17.png)

**Comment by janbijster** [comment]
@PedroDKE did you manage to improve the results? I'm experiencing a similar issue, also with data from librivox.

**Comment by PedroDKE** [comment]
@janbijster Yes & No. 
No as in : for this perticular audiobook/person i did not manage to improve the results. I think the audio/speech/microphone from this person might not be suited for TTS. Mainly the MFCC extractions did not look as expessive (if you compare my screens to others MFCC's you can see that my 'waves' are less expresive imo, not sure why this is as i havent looked much into what MFCC's really are).
Yes as in : When i used another book (other speaker, microphone etc) with much clearer/expressive MFCC's i was able to produce speech
